plugins/async-node/core/src/index.test.ts

```diff
@@ -340,3 +342,64 @@ test('replaces async nodes with chained multiNodes singular', async () => {
   expect(view?.actions[1].asset.type).toBe('text');
   expect(view?.actions[2].asset.type).toBe('text');
 });
+
+test('should call onAsyncNode hook when async node is encountered', async () => {
+  const plugin = new AsyncNodePlugin();
+
+  let deferredResolve: ((value: any) => void) | undefined;
+
+  // Call the tap method of the onAsyncNode hook
+ plugin.hooks.onAsyncNode.tap('test', async (node: Node.Async) => {
+    return new Promise((resolve) => {
+      deferredResolve = resolve;
```

Comment:
I don't think we'll really need to do anything with this `deferredResolve`, since we don't really need to resolve with any content later as we're just checking the node typing when the hook is first called.

Reply:
redone the entire test case